mantidproject/mantid
# Indirect - Data Analysis - IqtFit - Add calculated parameter values to property table in interface

The values of the fit parameters, calculated in IqtFit, are not currently used to update the property table in the IqtFit interface.
This PR adds the functionality for extracting the fit parameters from the \*\_Parameters workspace (generated by IqtFit) and using them to update the IqtFit interface.

**To test:**
1. Navigate to Interfaces > Indirect > Data Analysis > IqtFit.
2. Supply the \*\_iqt file included below into the 'Input' field.
3. For each optionin the 'Fit Type' drop-down menu (**Note:** Values greater than 1 or less than 0 will be ignored and not updated in the parameter table):
    1. Ensure the spectrum range defined in the 'Spectra Range' spinners covers the entire range of the input.
    2. Run the algorithm.
    3. Scroll through all values in the 'Plot Spectrum' spinner, checking the values in the property table against the values in the created \*\_Parameters workspace.
    4. Change the spectrum range to cover a sub-range of the input range.
    5. Run the algorithm.
    6. Scroll through all values in the 'Plot Spectrum' spinner, checking the values in the property table against the values in the created \*\_Parameters workspace. Where a spectrum is scrolled through, that is not within the sub-range used, check that the values are defaulted instead.
    7. Select a spectrum in the 'Plot Spectrum' spinner and click 'Fit Single Spectrum'.
    8. Scroll through all values in the 'Plot Spectrum' spin. Where the fit spectrum is scrolled through, check the value against the value in the created \*\_Parameters workspace.

**Test file:**
[iris26176_graphite002_iqt.zip](https://github.com/mantidproject/mantid/files/1333893/iris26176_graphite002_iqt.zip)



Fixes #20482. 

**Release Notes** 


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
